### ISSUE TITLE
Bugs when run ionic cordova run ios --device or use --target options

### ISSUE THREAD

**Issue by catnuxer** [ionic-team/ionic#12849]
> Ld /Users/nextg/Library/Developer/Xcode/DerivedData/Karyadi_GMDM-eknldtewijkwlcercyhmaotkpmjk/Build/Intermediates/ArchiveIntermediates/Karyadi\ GMDM/IntermediateBuildFilesPath/Karyadi\ GMDM.build/Debug-iphoneos/Karyadi\ GMDM.build/Objects-normal/arm64/Karyadi\ GMDM normal arm64
    cd /Users/nextg/Documents/Git/gmdm/platforms/ios
    export IPHONEOS_DEPLOYMENT_TARGET=9.1
    export PATH="/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr/bin:/Applications/Xcode.app/Contents/Developer/usr/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin"
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS10.3.sdk -L/Users/nextg/Documents/Git/gmdm/platforms/ios/build/device -F/Users/nextg/Documents/Git/gmdm/platforms/ios/build/device -filelist /Users/nextg/Library/Developer/Xcode/DerivedData/Karyadi_GMDM-eknldtewijkwlcercyhmaotkpmjk/Build/Intermediates/ArchiveIntermediates/Karyadi\ GMDM/IntermediateBuildFilesPath/Karyadi\ GMDM.build/Debug-iphoneos/Karyadi\ GMDM.build/Objects-normal/arm64/Karyadi\ GMDM.LinkFileList -Xlinker -rpath -Xlinker @executable_path/Frameworks -miphoneos-version-min=9.1 -dead_strip -Xlinker -object_path_lto -Xlinker /Users/nextg/Library/Developer/Xcode/DerivedData/Karyadi_GMDM-eknldtewijkwlcercyhmaotkpmjk/Build/Intermediates/ArchiveIntermediates/Karyadi\ GMDM/IntermediateBuildFilesPath/Karyadi\ GMDM.build/Debug-iphoneos/Karyadi\ GMDM.build/Objects-normal/arm64/Karyadi\ GMDM_lto.o -Xlinker -no_deduplicate -Xlinker -final_output -Xlinker /Applications/Karyadi\ GMDM.app/Karyadi\ GMDM -fobjc-arc -fobjc-link-runtime -ObjC /Users/nextg/Documents/Git/gmdm/platforms/ios/build/device/libCordova.a -weak_framework ImageIO -framework CoreLocation -framework AVFoundation -weak_framework AddressBook -weak_framework AddressBookUI -lz -weak_framework Social -weak_framework MessageUI -framework MediaPlayer -framework JavaScriptCore -Xlinker -dependency_info -Xlinker /Users/nextg/Library/Developer/Xcode/DerivedData/Karyadi_GMDM-eknldtewijkwlcercyhmaotkpmjk/Build/Intermediates/ArchiveIntermediates/Karyadi\ GMDM/IntermediateBuildFilesPath/Karyadi\ GMDM.build/Debug-iphoneos/Karyadi\ GMDM.build/Objects-normal/arm64/Karyadi\ GMDM_dependency_info.dat -o /Users/nextg/Library/Developer/Xcode/DerivedData/Karyadi_GMDM-eknldtewijkwlcercyhmaotkpmjk/Build/Intermediates/ArchiveIntermediates/Karyadi\ GMDM/IntermediateBuildFilesPath/Karyadi\ GMDM.build/Debug-iphoneos/Karyadi\ GMDM.build/Objects-normal/arm64/Karyadi\ GMDM
duplicate symbol _OBJC_CLASS_$_CDVLogger in:
    /Users/nextg/Library/Developer/Xcode/DerivedData/Karyadi_GMDM-eknldtewijkwlcercyhmaotkpmjk/Build/Intermediates/ArchiveIntermediates/Karyadi GMDM/IntermediateBuildFilesPath/Karyadi GMDM.build/Debug-iphoneos/Karyadi GMDM.build/Objects-normal/arm64/CDVLogger.o
    /Users/nextg/Documents/Git/gmdm/platforms/ios/build/device/libCordova.a(CDVLogger.o)
duplicate symbol _OBJC_METACLASS_$_CDVLogger in:
    /Users/nextg/Library/Developer/Xcode/DerivedData/Karyadi_GMDM-eknldtewijkwlcercyhmaotkpmjk/Build/Intermediates/ArchiveIntermediates/Karyadi GMDM/IntermediateBuildFilesPath/Karyadi GMDM.build/Debug-iphoneos/Karyadi GMDM.build/Objects-normal/arm64/CDVLogger.o
    /Users/nextg/Documents/Git/gmdm/platforms/ios/build/device/libCordova.a(CDVLogger.o)
ld: 2 duplicate symbols for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)

** ARCHIVE FAILED **

The following build commands failed:
	Ld /Users/nextg/Library/Developer/Xcode/DerivedData/Karyadi_GMDM-eknldtewijkwlcercyhmaotkpmjk/Build/Intermediates/ArchiveIntermediates/Karyadi\ GMDM/IntermediateBuildFilesPath/Karyadi\ GMDM.build/Debug-iphoneos/Karyadi\ GMDM.build/Objects-normal/arm64/Karyadi\ GMDM normal arm64
(1 failure)
Error: Error code 65 for command: xcodebuild with args: -xcconfig,/Users/nextg/Documents/Git/gmdm/platforms/ios/cordova/build-debug.xcconfig,-workspace,Karyadi GMDM.xcworkspace,-scheme,Karyadi GMDM,-configuration,Debug,-destination,generic/platform=iOS,-archivePath,Karyadi GMDM.xcarchive,archive,CONFIGURATION_BUILD_DIR=/Users/nextg/Documents/Git/gmdm/platforms/ios/build/device,SHARED_PRECOMPS_DIR=/Users/nextg/Documents/Git/gmdm/platforms/ios/build/sharedpch

> 

Did any body still in the same issues ?

**Comment by JakobMadsen717** [comment]
I have the same ld: 2 duplicate symbols for architecture arm64
problem when I do 
ionic cordova build ios --device
Building android has no problem

**Comment by dhavalsoni2001** [comment]
I am facing same issue while try to build.

**Comment by kensodemann** [comment]
Hello! Thank you for opening an issue with us! Would you be able to provide a sample application via GitHub that demonstrates the issue you are having? Thanks for using Ionic!

**Comment by JakobMadsen717** [comment]
I don't have an app I can easily share.

I can tell from manually building in xcode that it has something to do with @implementation CDVLogger
being included in two projects (CordovaLib and MyApp).

That in turn has something to do with Console feature being included twice in the auto-generated ./platforms/ios/MyApp/config.xml

    <feature name="Console">
        <param name="ios-package" value="CDVLogger" />
        <param name="onload" value="true" />
    </feature>

    ...

    <feature name="Console">
        <param name="ios-package" value="CDVLogger" />
    </feature>

That in turn has something to do with
<engine name="ios" spec="^4.5.0" /> and a related 
<plugin name="cordova-plugin-console" spec="1.0.5" />
perhaps (?), something I probably got updated in my project thru npm update

Anyway, for me a workaround is to: **ionic cordova plugin rm cordova-plugin-console**
That removed my duplicates and life is good building and running on ios.

As a side note, I _speculate_ you can reproduce the problem by creating a test app (ionic start BlahBlah), do npm update, which will perhaps do part of the damage, possibly do ionic cordova plugin add cordova-plugin-console unless already in there, and then I suspect that e.g. ionic cordova run ios --emulator would fail the build. But I am not certain.

**Comment by catnuxer** [comment]
Hi dude @kensodemann here I got error when already build and archive on Xcode.

I have try create simple ionic with my ionic id be739585 I start just simple create tabs. Then I try run on iOS and the result still in the same issues.

@JakobMadsen717  When you remove the cordova-plugin-console ? Is your apps so far so good run and build on Xcode to production ?

I also have problem when the engine-name="iOS" spec="^4.5.0" when run the ionic cordova platform add ios. But when I try with the ionic cordova platform add iOS @latest run well.

Did anybody have fixed this issue ?


![screen shot 2017-09-11 at 2 00 47 pm](https://user-images.githubusercontent.com/6625838/30262158-f6b92ef2-96f9-11e7-94a6-c0bc471208d4.png)

**Comment by PareshSolankiMI** [comment]
Same issue

**Comment by fwagner** [comment]
Given @JakobMadsen717's comment, I would assume this change is the has introduced the issue: https://github.com/apache/cordova-ios/commit/990d6d883d3891f57e9fd0ec5b8d7297070f590e. Telling npm to only use cordova-ios@4.4.0 in the package.json works for me and should be the easiest workaround for now.

**Comment by catnuxer** [comment]
Hi @fwagner  thank you for the good news. I already build and upload to production.

So the problem here is cordova-ios@latest.

And we could fixed this issues with:

1. Uninstall cordova-ios first, with npm uninstall cordova-ios
2. Install cordova-ios with npm install cordova-ios@4.4.0
3. After that we should add platform with ionic cordova platform ios@4.4.0
4. The last one, run ionic cordova build ios, that open with xcode and build with your settins.


Thank you guys.

Should we closed this issues ? or wait the ionic team to fixed cordova-ios@latest.

**Comment by BhawaniSingh** [comment]
@catnuxer thanks for the solution, it worked.

**Comment by kensodemann** [comment]
Hello! Thank you for opening an issue with us!

If I start a new app and do an `ionic cordova platform add ios`, it adds the appropriate version (`4.4.0`). Same for just `cordova platform add ios`

I will close this as it is not an issue with the Ionic Framework.

Thank you for using Ionic

**Comment by dhavalsoni2001** [comment]
When I am executing `ionic state reset` its installing cordova-ios version 4.5.0

**Comment by giltotherescue** [comment]
I followed @catnuxer's steps but am unfortunately still receiving the linker compile error. Here is my info. Any other ideas about how I might resolve this?

cli packages: ()

    @ionic/cli-utils  : 1.10.2
    ionic (Ionic CLI) : 3.10.3

global packages:

    Cordova CLI : 7.0.1 

local packages:

    @ionic/app-scripts : 2.1.3
    Cordova Platforms  : android 6.2.3 ios 4.4.0
    Ionic Framework    : ionic-angular 3.6.0

System:

    Android SDK Tools : 25.2.5
    ios-deploy        : 1.9.0 
    ios-sim           : 5.0.11 
    Node              : v6.9.4
    npm               : 4.6.1 
    OS                : macOS Sierra
    Xcode             : Xcode 8.3.3 Build version 8E3004b 


ld: 256 duplicate symbols for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)

** ARCHIVE FAILED **


The following build commands failed:
	Ld /Users/xander/Library/Developer/Xcode/DerivedData/Monkey_Method-alwnpqbcqnjvwadiqdfoxdyozcwx/Build/Intermediates/ArchiveIntermediates/Monkey\ Method/IntermediateBuildFilesPath/Monkey\ Method.build/Debug-iphoneos/Monkey\ Method.build/Objects-normal/arm64/Monkey\ Method normal arm64
(1 failure)
Error: Error code 65 for command: xcodebuild with args: -xcconfig,../platforms/ios/cordova/build-debug.xcconfig,-workspace,Monkey Method.xcworkspace,-scheme,Monkey Method,-configuration,Debug,-destination,generic/platform=iOS,-archivePath,Monkey Method.xcarchive,archive,CONFIGURATION_BUILD_DIR=../platforms/ios/build/device,SHARED_PRECOMPS_DIR=../platforms/ios/build/sharedpch

[ERROR] An error occurred while running cordova run ios (exit code 1).

**Comment by Richie765** [comment]
@giltotherescue, I had the same, this solved it for me:
```
ionic cordova platform remove ios
ionic cordova platform add ios@4.4.0
ionic cordova platform
# check that you really have 4.4.0 now
```

**Comment by giltotherescue** [comment]
Thanks @Richie765, I tried that and confirmed that it does show ios@4.4.0. Unfortunately the build still seems to be failing.

**Comment by giltotherescue** [comment]
After some trial and error, I discovered that cordova-plugin-ionic was causing my issue. After removing this plugin I was able to build. I will submit a new bug report with more details about the specific problem I ran into.

**Comment by jayj** [comment]
I fixed the issue by removing the `cordova-plugin-console` plugin after updating to cordova-ios@4.5.0. The plugin is [deprecated as the functionality provided is included in cordova-ios 4.5.0 or greater](https://github.com/apache/cordova-plugin-console/blob/master/README.md).

**Comment by ionut-movila** [comment]
To solve the problem please have a look over this blog post from Cordova iOS 4.5.
http://cordova.apache.org/announcements/2017/09/08/ios-release.html

You need to remove plugin "cordova-plugin-console" that is installed by default when generating a new Ionic project. Now this is in core part of cordova for iOS.

**Comment by melsayed008** [comment]
If you use ios 4.5 or higher you need to remove the plugin "cordova-plugin-console" because ios (platform) 4.5 or higher add this plugin by default so you end up with a duplicate error. Run the command "cordova plugin remove cordova-plugin-console" this should fix your problem. But doublecheck the file that this plugin doesn't exist in packages.json or config.xml. you can also run the command "cordova plugin ls" to check ur installed plugins.

**Comment by valerymadiedo** [comment]
Hello. I'm still having this issue with an ionic v1 project. I installed cocoa pods for the `phonegap-push-plugin`. I already tried removing `cordova-plugin-console`, but still the build fails with this error:

`duplicate symbol _OBJC_CLASS_$_AudioTogglePlugin in:
    /Users/vmadiedo/Library/Developer/Xcode/DerivedData/Poject-davisjelvfwxdwfymiborzjymhtd/Build/Intermediates.noindex/Poject.build/Debug-iphonesimulator/Poject.build/Objects-normal/x86_64/AudioTogglePlugin-728A025D894B7D7D.o
    /Users/vmadiedo/Library/Developer/Xcode/DerivedData/Poject-davisjelvfwxdwfymiborzjymhtd/Build/Intermediates.noindex/Poject.build/Debug-iphonesimulator/Poject.build/Objects-normal/x86_64/AudioTogglePlugin-36A809D7E8D28BA7.o

duplicate symbol _OBJC_METACLASS_$_AudioTogglePlugin in:
    /Users/vmadiedo/Library/Developer/Xcode/DerivedData/Poject-davisjelvfwxdwfymiborzjymhtd/Build/Poject.noindex/Poject.build/Debug-iphonesimulator/Poject.build/Objects-normal/x86_64/AudioTogglePlugin-728A025D894B7D7D.o
    /Users/vmadiedo/Library/Developer/Xcode/DerivedData/Poject-davisjelvfwxdwfymiborzjymhtd/Build/Poject.noindex/Poject.build/Debug-iphonesimulator/Poject.build/Objects-normal/x86_64/AudioTogglePlugin-36A809D7E8D28BA7.o
ld: 2 duplicate symbols for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)`

these are the plugins i have installed:

```
android-camera-permission 1.0.0 "Android Camera Permission"
com.dooble.audiotoggle 1.0.0 "AudioToggle"
cordova-custom-config 3.3.0 "cordova-custom-config"
cordova-plugin-ace 0.1.2 "Ace"
cordova-plugin-add-swift-support 1.7.0 "AddSwiftSupport"
cordova-plugin-android-permissions 0.10.0 "Permissions"
cordova-plugin-app-event 1.2.1 "Application Events"
cordova-plugin-audioinput 0.3.0 "Audio Input"
cordova-plugin-audiotoggle 1.0.3 "cordova-plugin-audiotoggle"
cordova-plugin-compat 1.2.0 "Compat"
cordova-plugin-contacts 2.3.1 "Contacts"
cordova-plugin-crosswalk-webview 2.3.0 "Crosswalk WebView Engine"
cordova-plugin-dbcopy 2.0.0 "sqlDB"
cordova-plugin-device 1.1.6 "Device"
cordova-plugin-dialogs 1.3.3 "Notification"
cordova-plugin-file 4.3.3 "File"
cordova-plugin-file-opener2 2.0.19 "File Opener2"
cordova-plugin-file-transfer 1.6.3 "File Transfer"
cordova-plugin-geolocation 2.4.3 "Geolocation"
cordova-plugin-google-analytics 1.8.3 "Google Universal Analytics Plugin"
cordova-plugin-inappbrowser 1.7.1 "InAppBrowser"
cordova-plugin-ionic-webview 1.1.16 "cordova-plugin-ionic-webview"
cordova-plugin-iosrtc 4.0.2 "iosrtc"
cordova-plugin-local-notifications-db 0.8.4 "LocalNotification"
cordova-plugin-nativegeocoder 1.0.2 "NativeGeocoder"
cordova-plugin-network-information 1.3.3 "Network Information"
cordova-plugin-screen-orientation 2.0.1 "Screen Orientation"
cordova-plugin-splashscreen 4.0.3 "Splashscreen"
cordova-plugin-statusbar 2.2.3 "StatusBar"
cordova-plugin-whitelist 1.3.2 "Whitelist"
cordova-sms-plugin 0.1.11 "Cordova SMS Plugin"
cordova-sqlite-storage 2.0.4 "Cordova sqlite storage plugin"
cordova.plugins.diagnostic 3.4.2 "Diagnostic"
es6-promise-plugin 4.1.0 "Promise"
ionic-plugin-keyboard 2.2.1 "Keyboard"
phonegap-plugin-push 2.1.0 "PushPlugin"
```
Here is my ionic info:

```
cli packages:
    @ionic/cli-plugin-cordova : 1.6.2 
    @ionic/cli-plugin-gulp    : 1.1.1 
    @ionic/cli-plugin-ionic1  : 2.1.1 
    @ionic/cli-utils          : 1.6.0 
    ionic (Ionic CLI)         : 3.6.0 
global packages:
    Cordova CLI : 7.1.0
    Gulp CLI    : CLI version 1.4.0 Local version 3.9.1
local packages:
    Cordova Platforms : android 6.3.0 ios 4.5.3
    Ionic Framework   : unknown
System:
    Node  : v6.11.3
    OS    : macOS High Sierra
    Xcode : Xcode 9.0 Build version 9A235
    npm   : 3.10.10
```

**Comment by VictorCoding** [comment]
@catnuxer you're the true mvp

**Comment by mariohmol** [comment]
Even using 4.5.x and without console plugin , this was not working. tryed many many times removing, adding, cleaning.. nothing.. the only way i could solve this was forcing 4.4.0 , so i guess still and issue

**Comment by jtomek** [comment]
> @melsayed008: But doublecheck the file that this plugin doesn't exist in **packages.json** or config.xml

This solved my problem. I can use ionic with ios 4.5.4 now. Thanks!

**Comment by jwburnside** [comment]
I ran into this issue with the latest ios version (~4.5.4), and did not have the cordova-plugin-console installed. Luckily had a coworker working on the same code base and we compared our setups. Turns out there were a couple of plugins I had that were unused, and one that had a different name structure than his. After I removed the unused plugins and re-added the misnamed one, the error went away. 
So if all else fails try removing your plugins and adding them back one-by-one to narrow down the offender.
